OneSignalSDK/onesignal/src/main/java/com/onesignal/ActivityLifecycleHandler.java

```diff
@@ -151,6 +161,41 @@ static private void logCurActivity() {
         OneSignal.Log(OneSignal.LOG_LEVEL.DEBUG, "curActivity is NOW: " + (curActivity != null ? "" + curActivity.getClass().getName() + ":" + curActivity : "null"));
     }
 
+    private static void logOrientationChange(int orientation) {
+        // Log device orientation change
+        if (orientation == Configuration.ORIENTATION_LANDSCAPE)
+            OneSignal.onesignalLog(OneSignal.LOG_LEVEL.DEBUG, "Configuration Orientation Change: LANDSCAPE (" + orientation + ")");
+        else if (orientation == Configuration.ORIENTATION_PORTRAIT)
+            OneSignal.onesignalLog(OneSignal.LOG_LEVEL.DEBUG, "Configuration Orientation Change: PORTRAIT (" + orientation + ")");
+    }
+
+    /**
+     * Takes pieces from onActivityResumed and onActivityStopped to recreate the view when the
+     * phones orientation is changed from manual detection using the onConfigurationChanged callback
+     * This fix was originally implemented for In App Messages not being re-shown when orientation
+     * was changed on wrapper SDK apps
+     */
+    private static void onOrientationChanged() {
+        // Remove view
+        handleLostFocus();
+        for (Map.Entry<String, ActivityAvailableListener> entry : sActivityAvailableListeners.entrySet()) {
+            entry.getValue().stopped(new WeakReference<>(curActivity));
+        }
+
+        // Show view
+        for (Map.Entry<String, ActivityAvailableListener> entry : sActivityAvailableListeners.entrySet()) {
+            entry.getValue().available(curActivity);
+        }
+
+        ViewTreeObserver treeObserver = curActivity.getWindow().getDecorView().getViewTreeObserver();
+        for (Map.Entry<String, OSSystemConditionController.OSSystemConditionObserver> entry : sSystemConditionObservers.entrySet()) {
+            KeyboardListener keyboardListener = new KeyboardListener(entry.getValue(), entry.getKey());
```

Comment:
Do we need to re setup the keyboard listener in the rotation case? Shouldn't the decorView be the same?

Reply:
The height issue with center modal occurs without resetting up the keyboard listeners

Reply:
ah gotcha, just want to make sure it was needed and didn't create issues with extra events